File: Analisis_data_shopee/shopee/datake2.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query = "SELECT*FROM order_detail"
df = pd.read_sql(query, connection)
connection.close()

print(df)

hasil = df.groupby('customer_id')['after_discount'].sum().reset_index(name = 'jumlah').sort_values('jumlah', ascending = False).reset_index(drop=True)
print(hasil)
print(hasil.iloc[1])

File: Analisis_data_shopee/shopee/datake2.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query <- "SELECT * FROM order_detail"
df <- dbGetQuery(con, query)
dbDisconnect(con)

print(head(df))

hasil <- df %>% group_by(customer_id) %>% summarise(jumlah = sum(after_discount)) %>% arrange(desc(jumlah))
print(hasil)
print(hasil[2, ])